File: python/doubly_linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
        self.prev = None

class DoublyLinkedList:
    def __init__(self):
        self.head = None

    def push(self, data):
        new_node = Node(data)
        new_node.next = self.head
        if self.head:
            self.head.prev = new_node
        self.head = new_node

    def display(self):
        curr = self.head
        while curr:
            print(curr.data, end=" <-> ")
            curr = curr.next
        print("None")

if __name__ == "__main__":
    dll = DoublyLinkedList()
    dll.push(30)
    dll.push(20)
    dll.push(10)
    dll.display()

File: java/DoublyLinkedList.java
public class DoublyLinkedList {
    Node head;
    class Node {
        int data;
        Node prev;
        Node next;
        Node(int d) { data = d; }
    }

    public void push(int new_data) {
        Node new_Node = new Node(new_data);
        new_Node.next = head;
        new_Node.prev = null;
        if (head != null) head.prev = new_Node;
        head = new_Node;
    }

    public void printlist(Node node) {
        Node last = null;
        while (node != null) {
            System.out.print(node.data + " ");
            last = node;
            node = node.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        DoublyLinkedList dll = new DoublyLinkedList();
        dll.push(6);
        dll.push(7);
        dll.push(1);
        dll.printlist(dll.head);
    }
}
